src/Config/RuntimeConfigLoader.cs

```diff
@@ -30,25 +31,77 @@ public abstract class RuntimeConfigLoader
     // state in place of using out params.
     public RuntimeConfig? RuntimeConfig;
 
+    public RuntimeConfigLoader(HotReloadEventHandler<HotReloadEventArgs>? handler = null, string? connectionString = null)
+    {
+        _handler = handler;
+        _connectionString = connectionString;
+    }
+
+    protected virtual void QueryManagerFactoryOnConfigChangedEvent(HotReloadEventArgs args)
+    {
+        _handler?.OnConfigChangedEvent(QUERY_MANAGER_FACTORY_ON_CONFIG_CHANGED, this, args);
+    }
+
+    protected virtual void MetadataProviderFactoryOnConfigChangedEvent(HotReloadEventArgs args)
+    {
+        _handler?.OnConfigChangedEvent(METADATA_PROVIDER_FACTORY_ON_CONFIG_CHANGED, this, args);
+    }
+
+    protected virtual void QueryEngineFactoryOnConfigChangedEvent(HotReloadEventArgs args)
+    {
+        _handler?.OnConfigChangedEvent(QUERY_ENGINE_FACTORY_ON_CONFIG_CHANGED, this, args);
+    }
+
+    protected virtual void MutationEngineFactoryOnConfigChangedEvent(HotReloadEventArgs args)
+    {
+        _handler?.OnConfigChangedEvent(MUTATION_ENGINE_FACTORY_ON_CONFIG_CHANGED, this, args);
+    }
```

Comment:
Seems like there's more opportunity for deduplicating code. Would this work:

Supplementing `HotReloadEventArgs` with the event name may enable you to consolidate the "*OnConfigChangedEvent" functions into one generic function `DabOnConfigChangedEvent(args)`.

And in that generic function, you'd do something specific based on the event name
```csharp
public void SendEventNotification(string message = "")
{
    HotReloadEventArgs args = new(message, eventName);
    GenericOnConfigChanged(new HotReloadEventArgs(eventName: DAB_EVENT_CONST_1, message: ""));
    GenericOnConfigChanged(new HotReloadEventArgs(eventName: DAB_EVENT_CONST_2, message: ""));
    GenericOnConfigChanged(new HotReloadEventArgs(eventName: DAB_EVENT_CONST_3, message: ""));
    /// etc
}

 protected virtual void GenericOnConfigChangedEvent(HotReloadEventArgs args)
{
    _handler?.OnConfigChangedEvent(this, args);
}
```

and in the HotReloadEventHandler class, pull the eventName out of the args object passed in:
```csharp

    public void OnConfigChangedEvent(object sender, TEventArgs args)
    {
        // use the arg.EventName property (I'm proposing that new property in the args class.)
        if (_eventHandlers.TryGetValue(args.eventName, out EventHandler<TEventArgs>? handler))
        {
            handler?.Invoke(sender, args);
        }
    }
```